### DISCUSSION TITLE
RT script updates to make UFS RT more flexible

### DISCUSSION THREAD

**Discussion by junwang-noaa** [ufs-community/ufs-weather-model#666] [General]
This is to extend the discussion in ufs-weather-model issue #656 to make UFS RT more flexible. Here are some thoughts to improve the UFS RT:

1) If specific namelist/model_confiure/nems.configure are used (e.g. input.nml for fv3), they will be listed in the test file tests/tests/specific_test.
2) the run script run_test.sh will not contain code to handle specific components, all the component related code in run_test.sh will go to a script run_comp.sh. run_comp.sh will check if specific component are set, if yes, required configuration files (input.nml for fv3, ww3_multi.inp for ww3 etc) will be generated.
3) parm directory will be reorganized, it is going to contain subdirectories including namelist, model_configure and nems_config directories, each contains  templates for that categories.
4) fv3_conf will be renamed (maybe prep_job) to avoid confusion. Under it, there will be subdirectory for script templates to prepare the input files in the run directory (e.g. control_run.IN) and job_card  directory with job card templates for all the supported platforms.
5) There will be some code clean up to use consistent the logic values (.T./.true.), single/double quotes, etc.

Hope these changes will make rt script cleaner and easy to follow. 
@uturuncoglu Please let us know if these helps resolving the issue you have when adding HAFS tests in UFS. Please share with us if you have other ideas on making the UFS RT more flexible.

**Comment by DeniseWorthen** [comment]
I would like to suggest that in a new design, we incorporate the ability to set the PE bounds automatically instead of having to hand edit multiple platforms/configurations in default_vars. I had opened this as an issue #320.

**Comment by arunchawla-NOAA** [comment]
Hi

I have not seen much discussion on this topic. Would be interested in seeing what ideas people have on this topic. Short term and long term how would you make rt.sh cleaner ?
